detectable/src/main/java/com/synopsys/integration/detectable/detectables/clang/compilecommand/CompileCommandParser.java

```diff
@@ -103,38 +104,74 @@ private String unEscapeDoubleQuotes(final String givenString) {
 
     private String escapeQuotedWhitespace(final String givenString) {
         final StringBuilder newString = new StringBuilder();
-        boolean lastCharWasEscapeChar = false;
-        boolean inQuotes = false;
-        boolean quoteTypeIsDouble = false;
+        final ParserState parserState = new ParserState();
         for (int i = 0; i < givenString.length(); i++) {
             final char c = givenString.charAt(i);
-            if (!inQuotes) {
-                if (!lastCharWasEscapeChar && (c == SINGLE_QUOTE_CHAR)) {
-                    inQuotes = true;
-                    quoteTypeIsDouble = false;
-                } else if (!lastCharWasEscapeChar && (c == DOUBLE_QUOTE_CHAR)) {
-                    inQuotes = true;
-                    quoteTypeIsDouble = true;
-                } else {
-                    newString.append(c);
-                }
+            if (!parserState.isInQuotes()) {
+                processNonQuotedChar(parserState, c, newString);
             } else {
```

Comment:
Small nitpick, but could we swap the if so we aren't negating the statement? I think it makes it easier to read.